corda/corda
# CORDA-3226 Fix dba migration for PostgreSQL following changes in CORDA-3009 and ENT-4192

Fixes for oid column type in node_identities:
CORDA-3009 Invalid hash function used for PersistentIdentity in PersistentIdentityService
ENT-4192 Identity service refactor for confidential-identities and accounts.

Test procedure:

1. Deploy any OS-4.1 Corda node (so pre OS-4.3 version).
2. In node.conf add:
```
dataSourceProperties = {
    dataSourceClassName = "org.postgresql.ds.PGSimpleDataSource"
    dataSource.url = "jdbc:postgresql://localhost:5432/postgres?currentSchema=my_schema"
    dataSource.user = my_user
    dataSource.password = "my_password"
}
database = {
    transactionIsolationLevel = READ_COMMITTED
}
```
3. Add `postgresql-42.2.6.jar` driver to drivers folder of the node.

4. Start PostgreSQL database and run on database:
```
CREATE USER "my_user" WITH LOGIN PASSWORD 'my_password';
CREATE SCHEMA "my_schema";
GRANT USAGE, CREATE ON SCHEMA "my_schema" TO "my_user";
GRANT SELECT, INSERT, UPDATE, DELETE, REFERENCES ON ALL tables IN SCHEMA "my_schema" TO "my_user";
ALTER DEFAULT privileges IN SCHEMA "my_schema" GRANT SELECT, INSERT, UPDATE, DELETE, REFERENCES ON tables TO "my_user";
GRANT USAGE, SELECT ON ALL sequences IN SCHEMA "my_schema" TO "my_user";
ALTER DEFAULT privileges IN SCHEMA "my_schema" GRANT USAGE, SELECT ON sequences TO "my_user";
ALTER ROLE "my_user" SET search_path = "my_schema";
```
5. Start node.

6. Stop node.

7. Replace corda.jar with the one under the test (OS-4.3-SNAPSHOT).

8. Run node.

9. Should start without errors.

10. Check in the database:
`node_identities.identity_value`
and `node_hash_to_key.identity_value` should have `bytea` column type.
